packages/webview_flutter/example/test_driver/webview_flutter_e2e.dart

```diff
@@ -828,6 +828,84 @@ void main() {
     expect(currentUrl, 'about:blank');
   });
+
+  testWidgets(
+    'can open new window and go back',
+    (WidgetTester tester) async {
+      final Completer<WebViewController> controllerCompleter =
+          Completer<WebViewController>();
+      final Completer<void> pageLoaded = Completer<void>();
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: WebView(
+            key: GlobalKey(),
+            onWebViewCreated: (WebViewController controller) {
+              controllerCompleter.complete(controller);
+            },
+            javascriptMode: JavascriptMode.unrestricted,
+            onPageFinished: (String url) {
+              pageLoaded.complete();
+            },
+            initialUrl: 'https://flutter.dev',
+          ),
+        ),
+      );
+      final WebViewController controller = await controllerCompleter.future;
+      await controller
+          .evaluateJavascript('window.open("https://www.google.com")');
+      await pageLoaded.future;
+      expect(controller.currentUrl(), completion('https://www.google.com/'));
+
+      await controller.goBack();
+      expect(controller.currentUrl(), completion('https://www.flutter.dev'));
+    },
+    skip: !Platform.isAndroid,
+  );
+
+  testWidgets(
+    'javascript does not run in parent window',
+    (WidgetTester tester) async {
+      final String openWindowTest = '''
+        <!DOCTYPE html><html>
+        <head><title>Resize test</title>
+          <script>
+            setTimeout(function() {
+                window.open('javascript:var elem = document.createElement("p");elem.innerHTML = "<b>Executed JS in parent origin: "+window.location.origin+"</b>"; document.body.append(elem);alert("XSS in doc.domain: "+document.domain+", win.origin: "+window.location.origin)');
+            }, 0);
+          </script>
+        </head>
+        <body onload="onLoad();" bgColor="blue">
+        </body>
+        </html>
+      ''';
+      final String openWindowTestBase64 =
+          base64Encode(const Utf8Encoder().convert(openWindowTest));
+      final Completer<WebViewController> controllerCompleter =
+          Completer<WebViewController>();
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: WebView(
+            key: GlobalKey(),
+            onWebViewCreated: (WebViewController controller) {
+              controllerCompleter.complete(controller);
+            },
+            javascriptMode: JavascriptMode.unrestricted,
+            initialUrl:
+                '<iframe src="data:text/html;charset=utf-8;base64,$openWindowTestBase64"</iframe>',
+          ),
+        ),
+      );
+
+      final WebViewController controller = await controllerCompleter.future;
+      final String result = await controller.evaluateJavascript(
+          'document.querySelector("p") && document.querySelector("p").textContent');
+      expect(result, isEmpty);
```

Comment:
This might be racy. 

How do we know that the code inside the iframe has run at this point? One solution is to add a `onLoad` handler to `iframe` and set some global state. e.g. `window.iframeLoaded = true`.  Then, at this point check that `window.iframeLoaded` is `true` and `document.querySelector("p") && document.querySelector("p").textContent` is empty.